src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -267,27 +273,30 @@ class SyncController(
       case PrintStatus =>
         val totalNodesCount = downloadedNodesCount + mptNodesQueue.size + nonMptNodesQueue.size
         log.info(
-          s"""|Block: ${appStateStorage.getBestBlockNumber()}/${targetBlock.number}.
+          s"""|Block: ${appStateStorage.getBestBlockNumber()}/${initialSyncState.targetBlock.number}.
               |Peers: ${assignedHandlers.size}/${handshakedPeers.size} (${blacklistedPeers.size} blacklisted).
               |State: $downloadedNodesCount/$totalNodesCount known nodes.""".stripMargin.replace("\n", " "))
     }
 
     def processSyncing(): Unit = {
       if (fullySynced) {
-        finish()
+        finishFastSync()
         startRegularSync()
       } else {
-        if (anythingQueued) processQueues()
+        if (anythingQueued || bestBlockHeaderNumber < initialSyncState.targetBlock.number) processQueues()
         else log.info("No more items to request, waiting for {} responses", assignedHandlers.size)
       }
     }
 
-    def finish(): Unit = {
+    def finishFastSync(): Unit = {
       syncStatePersistCancellable.cancel()
+      syncStateStorageActor ! PoisonPill
+
       fastSyncStateStorage.purge()
-      log.info("Fast sync finished")
       appStateStorage.fastSyncDone()
-      context.parent ! FastSyncDone
+
+      log.info("Fast sync finished")
+      context.parent ! FastSyncDone // TODO: is this still needed (seems to be used for BlockBroadcastActor)
```

Comment:
Yes, this was removed in https://github.com/input-output-hk/etc-client/pull/91 as they are no longer used